docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+
+Given below is an example usage scenario and how the delete mechanism behaves at each step.
+
+Step 1. The user decides to remove an `Entry` from the `EntryLog` and executes `/delete 1` command.
+
+_**Note:** The command will fail its execution if the index provided is invalid, and no `Entry` will be removed from the `EntryLog`. An error message will be displayed informing the user._
+
+Step 2. The command will be resolved by `Parser`, which would create an `DeleteCommmand` object containing the index of the `Entry` to be deleted.
+
+Step 3. When `execute()` method is called, a `Request` object is created.
+
+Step 4. From there, the `Request` is ready to be handled. `deleteEntry()` method is called and the `Entry` is removed from `EntryLog`.
+
+Step 5. A success message is after the `Entry` is removed from `EntryLog`.
+
+The following activity diagram summarizes what happens when a user executes a delete   command:
+
+![DeleteCommandActivityDiagram](static/frontend/commands/DeleteCommandActivityDiagram.png)
+
+<!-- @@author leonghuenweng -->
+#### Edit Command
+##### Overall class diagram for editing an Entry
+![EditCommandClassDiagram](static/frontend/commands/EditCommandClassDiagram.png)
+
+##### Implementation
+
+**Step 1.** User runs Edit command, specifying the ID of the entry to edit, as well as the new attributes of the respective
+fields.
+
+**Step 2.** Parser extracts the relevant argument and returns a EditCommand object.
+
+**Step 3.** The execute method of the EditCommand is called and a Request object is created. This request object specifies
+the modifications of the various fields.
+
+**Step 4.** The Request object is then parsed as an argument to the Backend Class, which feeds this request as an argument
+to a method in the EntryEndpoint class.
+
+**Step 5.** The EntryEndpoint class then finds and modifies the entry as specified by the user.
+
+**Step 6.** Upon successful completion of the modification, the EntryEndpoint class returns a Response object to the 
+Backend class. The Response object contains the updated fields of the entry.
+
+**Step 7.** The Backend class calls the save() method to update the Storage class with the edited entry.It then returns
+the Response object to the execute function in the EditCommand object.
+
+**Step 8.** The printExpenditureEdited method under the UI class is then called in the execute function and an
+acknowledgement message is printed to the user.
+
+##### Overall sequence diagram for editing an Entry
+
+![img_1.png](static/frontend/commands/EditCommandSequenceDiagram.png)
+
+#### View Command
+##### Class diagram of view command 
+![ViewCommandClassDiagram.png](./static/viewCommandClassDiagram.png)
+Class diagram above shows the methods called in the execute method of the 
+View Command object.
+
+##### Implementation
+![ViewCommandSequenceDiagram.png](static/frontend/commands/ViewCommandOverallSequenceDiagram.png)
```

Comment:
There is only one handing over of controls between ViewCommand and Backend, the use of sequence diagram may not be necessary here.